src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs

```diff
@@ -267,6 +267,7 @@ private RegionInfo GetCachedRegion(ICoreLogger logger)
                 return new RegionInfo(s_autoDiscoveredRegion, RegionAutodetectionSource.Cache, null);
             }
 
+            logger.Info($"[Region discovery] Auto-discovery successful but found null or empty region.");

```

Comment:
This is very subtle / unexpected. Removing a logging statemetn will break the test. 

Here's an example of how to mock the IMDS response: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/master/tests/Microsoft.Identity.Test.Unit/CoreTests/RegionDiscoveryProviderTests.cs#L290

This will be a much more realistic test, i.e. when we make an HTTP call to IMDS, the call takes a clong time.

Reply:
Added test